bootcd/rpms/genesis_scripts/src/root-bash_profile

```diff
@@ -1,14 +1,33 @@
 
-GENESIS_MODE="$(grep -o -e 'GENESIS_MODE=[^ ]*' /proc/cmdline | cut -c14-)"
-
-# setup the ruby environment we need
-source /etc/profile.d/rvm.sh
-rvm list
-ruby --version
-
-if [[ ! -z $GENESIS_MODE ]] && [ ! -e "/var/run/bootloader-autorun.lock" ]
-  then
-    touch /var/run/bootloader-autorun.lock
-    # disable task execution prompting when doing initial target run
-    GENESIS_PROMPT_TIMEOUT=0  /usr/bin/genesis-bootloader 2>&1 | tee /var/log/genesis-bootloader.log
+# Genesis modifications
+
+# Get our configuration
+source /etc/sysconfig/genesis
+# If the config file is broken, use sane defaults
+GENESIS_PIDFILE=${GENESIS_PIDFILE:-/var/run/genesis.pid}
+GENESIS_LOGFILE=${GENESIS_LOGFILE:-/var/log/genesis-bootloader.log}
+
+# Try for two seconds to find a pidfile
+for i in {1..20}
+do
+	if [[ -f "${GENESIS_PIDFILE}" ]]
+	then
+		# get the pid and break out of the for loop
+		GENESIS_PID=$(cat "${GENESIS_PIDFILE}")
```

Comment:
I was thinking something like http://refspecs.linuxbase.org/LSB_3.0.0/LSB-PDA/LSB-PDA/iniscrptfunc.html pidofproc, but no worries if you do t want to use that :)